### PR TITLE
docs/options-in-versions: which version added each cmdline option

### PR DESCRIPTION
**Pull request by bagder**
Added test 971 to verify that the list is in sync with the files in
cmdline-opts. The check also verifies that .d-files that uses Added:
specify the same version number as the options-in-versions file does.

### CODE REVIEW

**Comment by mback2k** [comment]
Just FYI: this new test case does not seem to work on Windows yet, but I don't have time to look deeper today.

File: tests/options-scan.pl [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        }
+    }
+    return @options;
+}
+
+sub versioncheck {
+    my ($f, $v)=@_;
+    open(F, "<$cmddir/$f.d");
+    while(<F>) {
+        if(/^Added: (.*)/) {
```
Comment (Lekensteyn) [inline tests/options-scan.pl:78]: Perhaps CR from CRLFs are being matched here? Try:
```suggestion
        if(/^Added: ([0-9.]+)/) {
```

File: tests/options-scan.pl [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            if($1 ne $v) {
+                print STDERR "$f lists $v in doc but $1 in file\n";
+                $error++;
+            }
+            last;
+        }
+    }
+    close(F);
```
Comment (Lekensteyn) [inline tests/options-scan.pl:86]: Is a check needed to ensure that the `Version:` line was present and correctly formatted?

**Comment by Lekensteyn** [comment]
The message `perl -I.  returned 122, when expecting 0` looks somewhat truncated,
though the script does seem to run. My guess is that the regex is too lenient.

<details><summary>test 0971...[Verify that options-in-versions and docs/cmdline-opts are in sync]</summary>

```
perl -I.  returned 122, when expecting 0
 exit FAILED
== Contents of files in the log/ dir after test 971
=== Start of file commands.log
 perl -I. ./options-scan.pl ./../docs/options-in-versions ./../docs/cmdline-opts >log/stdout971 2>log/stderr971
=== End of file commands.log
=== Start of file ftpserver.cmd
 Testnum 971
=== End of file ftpserver.cmd
=== Start of file stderr971
 abstract-unix-socket lists 7.53.0 in doc but 7.53.0
  in file
 alt-svc lists 7.64.1 in doc but 7.64.1
  in file
 cert-status lists 7.41.0 in doc but 7.41.0
  in file
 compressed-ssh lists 7.56.0 in doc but 7.56.0
  in file
 connect-to lists 7.49.0 in doc but 7.49.0
  in file
 crlfile lists 7.19.7 in doc but 7.19.7
  in file
 data-raw lists 7.43.0 in doc but 7.43.0
  in file
 data-urlencode lists 7.18.0 in doc but 7.18.0
  in file
 disallow-username-in-url lists 7.61.0 in doc but 7.61.0
  in file
 dns-interface lists 7.33.0 in doc but 7.33.0
  in file
 dns-ipv4-addr lists 7.33.0 in doc but 7.33.0
  in file
 dns-ipv6-addr lists 7.33.0 in doc but 7.33.0
  in file
 dns-servers lists 7.33.0 in doc but 7.33.0
  in file
 doh-url lists 7.62.0 in doc but 7.62.0
  in file
 etag-compare lists 7.68.0 in doc but 7.68.0
  in file
 etag-save lists 7.68.0 in doc but 7.68.0
  in file
 expect100-timeout lists 7.47.0 in doc but 7.47.0
  in file
 fail-early lists 7.52.0 in doc but 7.52.0
  in file
 false-start lists 7.42.0 in doc but 7.42.0
  in file
 ftp-account lists 7.13.0 in doc but 7.13.0
  in file
 ftp-alternative-to-user lists 7.15.5 in doc but 7.15.5
  in file
 ftp-method lists 7.15.1 in doc but 7.15.1
  in file
 ftp-pasv lists 7.11.0 in doc but 7.11.0
  in file
 ftp-pret lists 7.20.0 in doc but 7.20.0
  in file
 ftp-skip-pasv-ip lists 7.14.2 in doc but 7.14.2
  in file
 ftp-ssl-ccc lists 7.16.1 in doc but 7.16.1
  in file
 ftp-ssl-ccc-mode lists 7.16.2 in doc but 7.16.2
  in file
 ftp-ssl-control lists 7.16.0 in doc but 7.16.0
  in file
 happy-eyeballs-timeout-ms lists 7.59.0 in doc but 7.59.0
  in file
 haproxy-protocol lists 7.60.0 in doc but 7.60.0
  in file
 hostpubmd5 lists 7.17.1 in doc but 7.17.1
  in file
 http1.1 lists 7.33.0 in doc but 7.33.0
  in file
 http2 lists 7.33.0 in doc but 7.33.0
  in file
 http2-prior-knowledge lists 7.49.0 in doc but 7.49.0
  in file
 http3 lists 7.66.0 in doc but 7.66.0
  in file
 keepalive-time lists 7.18.0 in doc but 7.18.0
  in file
 libcurl lists 7.16.1 in doc but 7.16.1
  in file
 list-only lists 4.0 in doc but 4.0
  in file
 local-port lists 7.15.2 in doc but 7.15.2
  in file
 login-options lists 7.34.0 in doc but 7.34.0
  in file
 mail-auth lists 7.25.0 in doc but 7.25.0
  in file
 mail-from lists 7.20.0 in doc but 7.20.0
  in file
 mail-rcpt lists 7.20.0 in doc but 7.20.0
  in file
 mail-rcpt-allowfails lists 7.69.0 in doc but 7.69.0
  in file
 metalink lists 7.27.0 in doc but 7.27.0
  in file
 netrc-file lists 7.21.5 in doc but 7.21.5
  in file
 next lists 7.36.0 in doc but 7.36.0
  in file
 no-alpn lists 7.36.0 in doc but 7.36.0
  in file
 no-npn lists 7.36.0 in doc but 7.36.0
  in file
 no-progress-meter lists 7.67.0 in doc but 7.67.0
  in file
 no-sessionid lists 7.16.0 in doc but 7.16.0
  in file
 noproxy lists 7.19.4 in doc but 7.19.4
  in file
 parallel lists 7.66.0 in doc but 7.66.0
  in file
 parallel-immediate lists 7.68.0 in doc but 7.68.0
  in file
 parallel-max lists 7.66.0 in doc but 7.66.0
  in file
 path-as-is lists 7.42.0 in doc but 7.42.0
  in file
 post301 lists 7.17.1 in doc but 7.17.1
  in file
 post302 lists 7.19.1 in doc but 7.19.1
  in file
 post303 lists 7.26.0 in doc but 7.26.0
  in file
 preproxy lists 7.52.0 in doc but 7.52.0
  in file
 proto lists 7.20.2 in doc but 7.20.2
  in file
 proto-default lists 7.45.0 in doc but 7.45.0
  in file
 proto-redir lists 7.20.2 in doc but 7.20.2
  in file
 proxy-anyauth lists 7.13.2 in doc but 7.13.2
  in file
 proxy-cacert lists 7.52.0 in doc but 7.52.0
  in file
 proxy-capath lists 7.52.0 in doc but 7.52.0
  in file
 proxy-cert lists 7.52.0 in doc but 7.52.0
  in file
 proxy-cert-type lists 7.52.0 in doc but 7.52.0
  in file
 proxy-ciphers lists 7.52.0 in doc but 7.52.0
  in file
 proxy-crlfile lists 7.52.0 in doc but 7.52.0
  in file
 proxy-header lists 7.37.0 in doc but 7.37.0
  in file
 proxy-insecure lists 7.52.0 in doc but 7.52.0
  in file
 proxy-key-type lists 7.52.0 in doc but 7.52.0
  in file
 proxy-negotiate lists 7.17.1 in doc but 7.17.1
  in file
 proxy-pass lists 7.52.0 in doc but 7.52.0
  in file
 proxy-service-name lists 7.43.0 in doc but 7.43.0
  in file
 proxy-ssl-allow-beast lists 7.52.0 in doc but 7.52.0
  in file
 proxy-tlsauthtype lists 7.52.0 in doc but 7.52.0
  in file
 proxy-tlspassword lists 7.52.0 in doc but 7.52.0
  in file
 proxy-tlsuser lists 7.52.0 in doc but 7.52.0
  in file
 proxy-tlsv1 lists 7.52.0 in doc but 7.52.0
  in file
 raw lists 7.16.2 in doc but 7.16.2
  in file
 remote-name-all lists 7.19.0 in doc but 7.19.0
  in file
 request-target lists 7.55.0 in doc but 7.55.0
  in file
 resolve lists 7.21.3 in doc but 7.21.3
  in file
 retry lists 7.12.3 in doc but 7.12.3
  in file
 retry-all-errors lists 7.71.0 in doc but 7.71.0
  in file
 retry-connrefused lists 7.52.0 in doc but 7.52.0
  in file
 retry-delay lists 7.12.3 in doc but 7.12.3
  in file
 retry-max-time lists 7.12.3 in doc but 7.12.3
  in file
 sasl-authzid lists 7.66.0 in doc but 7.66.0
  in file
 sasl-ir lists 7.31.0 in doc but 7.31.0
  in file
 service-name lists 7.43.0 in doc but 7.43.0
  in file
 socks4 lists 7.15.2 in doc but 7.15.2
  in file
 socks4a lists 7.18.0 in doc but 7.18.0
  in file
 socks5 lists 7.18.0 in doc but 7.18.0
  in file
 socks5-basic lists 7.55.0 in doc but 7.55.0
  in file
 socks5-gssapi lists 7.55.0 in doc but 7.55.0
  in file
 socks5-gssapi-nec lists 7.19.4 in doc but 7.19.4
  in file
 socks5-gssapi-service lists 7.19.4 in doc but 7.19.4
  in file
 socks5-hostname lists 7.18.0 in doc but 7.18.0
  in file
 ssl lists 7.20.0 in doc but 7.20.0
  in file
 ssl-allow-beast lists 7.25.0 in doc but 7.25.0
  in file
 ssl-no-revoke lists 7.44.0 in doc but 7.44.0
  in file
 ssl-reqd lists 7.20.0 in doc but 7.20.0
  in file
 ssl-revoke-best-effort lists 7.70.0 in doc but 7.70.0
  in file
 styled-output lists 7.61.0 in doc but 7.61.0
  in file
 tcp-fastopen lists 7.49.0 in doc but 7.49.0
  in file
 tcp-nodelay lists 7.11.2 in doc but 7.11.2
  in file
 tftp-blksize lists 7.20.0 in doc but 7.20.0
  in file
 tftp-no-options lists 7.48.0 in doc but 7.48.0
  in file
 tls-max lists 7.54.0 in doc but 7.54.0
  in file
 tlsauthtype lists 7.21.4 in doc but 7.21.4
  in file
 tlspassword lists 7.21.4 in doc but 7.21.4
  in file
 tlsuser lists 7.21.4 in doc but 7.21.4
  in file
 tlsv1.0 lists 7.34.0 in doc but 7.34.0
  in file
 tlsv1.1 lists 7.34.0 in doc but 7.34.0
  in file
 tlsv1.2 lists 7.34.0 in doc but 7.34.0
  in file
 tlsv1.3 lists 7.52.0 in doc but 7.52.0
  in file
 tr-encoding lists 7.21.6 in doc but 7.21.6
  in file
 trace-time lists 7.14.0 in doc but 7.14.0
  in file
 unix-socket lists 7.40.0 in doc but 7.40.0
  in file
=== End of file stderr971
```
</details>

**Comment by bagder** [comment]
Isn't it strange that test 971 now only fails on the windows machines on azure, the other windows jobs work...

**Comment by bagder** [comment]
BTW: this document lists all **currently available** options and it is not a representation of all options that ever existed during the project lifetime. This has the effect that for example curl 4.0 (probably) had 21 options but in this list there are only 18 mentioned for `4.0` - this is because three of those early options were subsequently later removed/replaced by other options. This should be kept in mind if you for example compare this data with the dashboard graph that shows a graph of number of cmdline options over time.

**Pull request closed without merging**